src/filters/body.rs

```diff
@@ -105,8 +105,19 @@ pub fn stream() -> impl Filter<Extract = (BodyStream,), Error = Rejection> + Cop
 ///     });
 /// ```
 pub fn concat() -> impl Filter<Extract = (FullBody,), Error = Rejection> + Copy {
-    body().and_then(|body: ::hyper::Body| Concat {
-        fut: body.try_concat(),
+    stream().and_then(|mut body_stream: BodyStream| async move {
+        use bytes::{BytesMut, Bytes};
+        use futures::stream::*;
+        let mut bytes = BytesMut::new();
+        loop {
+            match body_stream.next().await {
+                Some(Ok(data)) => bytes.extend_from_slice(data.bytes()),
```

Comment:
Oh yea huh, I think `Bytes::extend(Bytes)` can't use specialization like `Vec` can, and so it's much slower. :cry:

Reply:
True, Ideally, I would like to read the content length header, if any, and allocate a Vec::with_capacity, and then try to re-use that allocation somehow in the FullBody. What would be the best way to get the content length?

Reply:
I figured out how to do it, ingenious filter chain architecture :) available on on karlri:concat-performance